mtg_card_identifier/CardScanner/CardScannerView.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  CardScannerView.swift
+//  mtg_card_identifier
+//
+//  Created by mac on 8/16/23.
+//
+
+import SwiftUI
+import VisionKit
+import AVFoundation
+
+struct CardScannerView: View {
+    @State private var cameraAccessAuthorized = false
+
+    var body: some View {
+        NavigationView {
+            if cameraAccessAuthorized {
+                CardScanner()
+            } else {
+                NoCameraAccessView()
+            }
+        }
+        .onAppear {
+            requestCameraAccess()
+            checkCameraAuthorizationStatus()
+        }
+    }
+
+    func requestCameraAccess() {
+        AVCaptureDevice.requestAccess(for: .video) { granted in
+            if granted {
+                print("Camera access granted")
+                cameraAccessAuthorized = true
+            } else {
+                print("Camera access denied")
+                cameraAccessAuthorized = false
+            }
+        }
+    }
+    
+    func checkCameraAuthorizationStatus() {
+            let authorizationStatus = AVCaptureDevice.authorizationStatus(for: .video)
+            
+            switch authorizationStatus {
+            case .authorized:
+                cameraAccessAuthorized = true
+            case .notDetermined, .denied, .restricted:
+                cameraAccessAuthorized = false
+            @unknown default:
+                cameraAccessAuthorized = false
+            }
+        }
+    }
+
+struct CardScannerView_Previews: PreviewProvider {
+    static var previews: some View {
+        CardScannerView()
+    }
+}
+
+
+struct CardScanner: View {
+    @State private var scanedCard = [""]
+//    @State private var style: Style? = nil
+    var body: some View {
+        NavigationStack {
+            VStack {
+                LiveTextScanner(scanedText: $scanedCard, style: Style.cardShape)
+            }
+            .toolbar {
+                ToolbarItem {
+                    Button() {
+                        
```

Comment:
This will eventually be TCA. I think we do that in a different PR, but immediately after this one.